geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/io/fs/LocalDelegate.scala

```diff
@@ -57,9 +60,27 @@ class LocalDelegate extends FileSystemDelegate {
 }
 
 object LocalDelegate {
+
   class LocalFileHandle(file: File) extends FileHandle {
     override def path: String = file.getAbsolutePath
     override def length: Long = file.length()
     override def open: InputStream = new FileInputStream(file)
   }
+
+  class StdInHandle extends FileHandle {
+    private val is: CountingInputStream = new CountingInputStream(System.in)
+    private val opened = new AtomicBoolean(false)
```

Comment:
should open default to false if you are opening a new input stream above? seems that you could call length before calling open?

Reply:
this is a bit of a hack to shoehorn `stdin` to the existing file-based model - in order to track the length read (used for % complete in the ingest command), i wanted to keep a single instance of the CountingInputStream. I think i'll move that logic to the ingest command though, and not have it here, since it doesn't quite fit with the `FileHandle` api.